### DISCUSSION TITLE
Does U8G2 set SPI clock pin (SCK)?

### DISCUSSION THREAD

**Discussion by Fishbone69** [olikraus/u8g2#2236] [General]
Hi:
     The default SPI clock pin for esp32-C3 is GPIO6 according to the datasheet.  However, when I checked this running an app with u8g2 it showed SPI clock was on GPIO2 (I havent run it without u8g2 - I plan to do that in a bit).  I was wondering if the constructor where some of the pins are set also determines the clk pin?

I'm asking because I am learning about hardware vs software SPI and looking at the IO_MUX  and GPIO matrix on esp32 and tring to understand the differences.

**Comment by olikraus** [comment]
U8g2 will just use the Arduino SPI  subsystem without modification. So whatever is defines as SPI clock pin for your Arduino implementation on your board will be used by u8g2.

**Comment by Fishbone69** [comment]
I figured but just wanted to be sure.Thanks!
-------- Original message --------From: olikraus ***@***.***> Date: 8/4/23  1:42 PM  (GMT-06:00) To: olikraus/u8g2 ***@***.***> Cc: Fishbone69 ***@***.***>, Author ***@***.***> Subject: Re: [olikraus/u8g2] Does U8G2 set SPI clock pin (SCK)? (Discussion #2236) 
U8g2 will just use the Arduino SPI  subsystem without modification. So whatever is defines as SPI clock pin for your Arduino implementation on your board will be used by u8g2.

**Comment by Fishbone69** [comment]
Per the pins_arduino.h - U8g2 is using the correct GPIO.

CLOSED
